pulsar-broker/src/test/java/org/apache/pulsar/client/api/PatternConsumerTopicWatcherBackPressureMultipleConsumersTest.java

```diff
@@ -0,0 +1,199 @@
+                            threadPoolConfig -> threadPoolConfig.numberOfThreads(2))
+                    .build();
+            List<PulsarClientImpl> clients = new ArrayList<>(numberOfClients);
+            @Cleanup
+            Closeable closeClients = () -> {
+                for (PulsarClient client : clients) {
+                    try {
+                        client.close();
+                    } catch (PulsarClientException e) {
+                        log.error("Failed to close client {}", client, e);
+                    }
+                }
+            };
+            for (int i = 0; i < numberOfClients; i++) {
+                PulsarClientImpl client = (PulsarClientImpl) PulsarClient.builder()
+                        .serviceUrl(getClientServiceUrl())
+                        .statsInterval(0, TimeUnit.SECONDS)
+                        .operationTimeout(1, TimeUnit.MINUTES)
+                        .sharedResources(sharedResources)
+                        .build();
+                clients.add(client);
+            }
+
+            List<CompletableFuture<Consumer<String>>> consumerFutures = new ArrayList<>(numberOfClients);
+            for (int i = 0; i < topicCount; i++) {
+                String topicsPattern = namespace + "/a+-" + i + "_[01]$";
+                CompletableFuture<Consumer<String>> consumerFuture =
+                        clients.get(i % numberOfClients).newConsumer(Schema.STRING)
+                                .topicsPattern(topicsPattern).subscriptionName("sub" + i)
+                                .subscribeAsync();
+                consumerFutures.add(consumerFuture);
+                consumerFuture.exceptionally(throwable -> {
+                    log.error("Failed to subscribe to pattern {}", topicsPattern, throwable);
+                    return null;
+                });
+            }
+
+            FutureUtil.waitForAll(consumerFutures).get(60, TimeUnit.SECONDS);
+
+            List<Consumer<String>> consumers = consumerFutures.stream().map(CompletableFuture::join).toList();
+
+            PulsarClientImpl client = clients.get(0);
+            sendAndValidate(topicCount, client, consumers, topicNamePrefix, "_0");
+
+            // create additional topics
+            createTopics(topicCount, topicNamePrefix, "_1");
+
+            // send to additional topic
+            sendAndValidate(topicCount, client, consumers, topicNamePrefix, "_1");
+        }
+
+        validateThatTokensHaventLeakedOrIncreased();
+    }
+
+    protected void validateThatTokensHaventLeakedOrIncreased() {
+        AsyncDualMemoryLimiterImpl limiter =
+                pulsar.getBrokerService().getMaxTopicListInFlightLimiter();
+        assertThat(limiter.getLimiter(AsyncDualMemoryLimiter.LimitType.HEAP_MEMORY).getAvailablePermits())
+                .isEqualTo(pulsar.getConfiguration().getMaxTopicListInFlightHeapMemSizeMB() * 1024 * 1024);
```

Comment:
If `getMaxTopicListInFlightHeapMemSizeMB()` or `getMaxTopicListInFlightDirectMemSizeMB()` returns a value of 2048 or more, the expression `* 1024 * 1024` will overflow a 32-bit integer. To prevent this, use `1024L` to ensure the computation is performed with long integers.

```diff
-        .isEqualTo(pulsar.getConfiguration().getMaxTopicListInFlightHeapMemSizeMB() * 1024 * 1024);
+        .isEqualTo(pulsar.getConfiguration().getMaxTopicListInFlightHeapMemSizeMB() * 1024L * 1024);
-        .isEqualTo(pulsar.getConfiguration().getMaxTopicListInFlightDirectMemSizeMB() * 1024 * 1024);
+        .isEqualTo(pulsar.getConfiguration().getMaxTopicListInFlightDirectMemSizeMB() * 1024L * 1024);
```